cmd/util/flags/configflags/job_selection.go

```diff
@@ -1,43 +1,40 @@
 package configflags
 
-import "github.com/bacalhau-project/bacalhau/pkg/config/types"
-
-//
-// Requester Job Selection Policy
-//
-// TODO: https://github.com/bacalhau-project/bacalhau/issues/2929
-// The flags are currently settable on the requester, however the requester does not support a JobSelectionPolicy
-// setting the flags is a noop until the above issue is fixed.
+import (
+	"github.com/bacalhau-project/bacalhau/pkg/config/types"
+)
 
 var JobSelectionFlags = []Definition{
 	{
-		FlagName:     "job-selection-data-locality",
-		ConfigPath:   types.NodeComputeJobSelectionLocality,
-		DefaultValue: Default.Node.Compute.JobSelection.Locality,
-		Description:  `Only accept jobs that reference data we have locally ("local") or anywhere ("anywhere").`,
+		FlagName:          "job-selection-data-locality",
+		ConfigPath:        "job.selection.data.locality.deprecated",
+		DefaultValue:      "",
+		FailIfUsed:        true,
+		Deprecated:        true,
+		DeprecatedMessage: "Locality is no longer configurable.",
```

Comment:
My concern here is that we are deprecating a flag that no longer corresponds to a field in the config, and in this case no longer corresponds to a feature in general. This means that the flag will effectively become a no-op, and can lead to confusion for users because they might provide the flag expecting it to influence the behavior of the node, but it won't do anything. Are we saying that it's okay to start the node anyways?